pkg/operation/botanist/botanist.go

```diff
@@ -165,6 +166,22 @@ func (b *Botanist) CreateETCDSnapshot(ctx context.Context) error {
 	return err
 }
 
+// UpdateShootAndCluster updates the given `core.gardener.cloud/v1beta1.Shoot` resource in the garden cluster after
+// applying the given transform function to it. It will also update the `shoot` field in the
+// extensions.gardener.cloud/v1alpha1.Cluster` resource in the seed cluster with the updated shoot information.
+func (b *Botanist) UpdateShootAndCluster(ctx context.Context, shoot *gardencorev1beta1.Shoot, transform func() error) error {
+	if _, err := controllerutil.CreateOrUpdate(ctx, b.K8sGardenClient.Client(), shoot, transform); err != nil {
```

Comment:
Shouldn't this be updated with a "retry on conflict"?
Also, we should probably use the direct client for the Shoot update, as we are likely to run into conflicts otherwise, as we are constantly updating the status in the shoot flow.
```suggestion
	if err := kutil.TryUpdate(ctx, retry.DefaultRetry, b.K8sGardenClient.DirectClient(), shoot, transform); err != nil {
```

Reply:
done